### Proof of testing

Regression test fails without the fix and passes with it:

```
# base.py reverted to master (no bounds guard)
$ pytest test/messages/test_message_base.py::test_extract_text_and_code_prompts_unclosed_fence -q
E   IndexError: list index out of range
1 failed in 0.40s

# with the fix
$ pytest test/messages/test_message_base.py::test_extract_text_and_code_prompts_unclosed_fence -q
1 passed in 0.36s
```

`ruff check` and `ruff format --check` are clean on both changed files.

### Note on the failing CI checks

The `pytest_*` jobs fail at **collection time**, not on this change:

```
ERROR test/agents/test_chat_agent.py - ValueError: Missing or empty required
API keys in environment variables: OPENAI_API_KEY.
Process completed with exit code 2
```

Because this PR comes from a fork, GitHub does not expose the repository Secrets, so `OPENAI_API_KEY` (and the other keys) are empty and the agent-instantiating tests error on import before any test runs. The `pre-commit` check passes. Happy to re-run under a maintainer-triggered workflow that has Secrets access.

--- a/camel/messages/base.py
+++ b/camel/messages/base.py
@@ -297,6 +297,10 @@ class BaseMessage:
             Tuple[List[TextPrompt], List[CodePrompt]]: A tuple containing a
                 list of text prompts and a list of code prompts extracted
                 from the content.
+
+        Raises:
+            ValueError: If the content contains a code fence that is opened
+                but never closed (for example, truncated model output).
         """
         text_prompts: List[TextPrompt] = []
         code_prompts: List[CodePrompt] = []
@@ -322,6 +326,8 @@ class BaseMessage:
                 not lines[idx].lstrip().startswith("```")
             ):
                 idx += 1
+            if idx >= len(lines):
+                raise ValueError("Unclosed code fence in message content")
             code = "\n".join(lines[start_idx:idx]).strip()
             code_prompts.append(CodePrompt(code, code_type=code_type))
 

--- a/test/messages/test_message_base.py
+++ b/test/messages/test_message_base.py
@@ -87,22 +87,19 @@ def test_extract_text_and_code_prompts():
 
 def test_extract_text_and_code_prompts_unclosed_fence():
     # A code block whose closing fence is missing (e.g. truncated model
-    # output) must not raise; the remainder is treated as the code body.
+    # output) must fail explicitly rather than silently treating the
+    # remainder as a valid code prompt, so callers can handle the malformed
+    # input intentionally.
     base_message = BaseMessage(
         role_name="test_role_name",
         role_type=RoleType.USER,
         meta_dict=dict(),
         content="This is a text prompt.\n\n```python\nprint('unclosed')",
     )
-    text_prompts, code_prompts = base_message.extract_text_and_code_prompts()
-
-    assert len(text_prompts) == 1
-    assert text_prompts[0] == "This is a text prompt."
-
-    assert len(code_prompts) == 1
-    assert isinstance(code_prompts[0], CodePrompt)
-    assert code_prompts[0] == "print('unclosed')"
-    assert code_prompts[0].code_type == "python"
+    with pytest.raises(
+        ValueError, match="Unclosed code fence in message content"
+    ):
+        base_message.extract_text_and_code_prompts()
 
 
 def test_base_message_to_dict(base_message: BaseMessage) -> None:

